controller/pods.go

```diff
@@ -142,24 +164,42 @@ func (c *Controller) delNetwork(podObj *apiv1.Pod, networkName string, np cniPod
 	if np.IsPrimary {
 		return nil
 	}
+	glog.V(5).Infof("Deleting network %s from pod %s", networkName, podObj.ObjectMeta.Name)
 	cniParams, err := c.getCNIParams(podObj, networkName, np)
 	if err != nil {
 		return err
 	}
 
-	for i := 0; i < maxRetries; i++ {
-		err = c.cniPlugin.DeleteNetwork(cniParams)
-		if err == nil {
-			if i > 0 {
-				glog.V(4).Infof("Succeeded deleting network %s on Pod %s after %d attempt", networkName, podObj.ObjectMeta.Name, i)
+	err = c.cniPlugin.DeleteNetwork(cniParams)
+	if err != nil {
+		go c.retryDelNetwork(podObj.ObjectMeta.Name, networkName, cniParams)
+	}
+	return err
+}
+
+func (c *Controller) podAdded(newObj interface{}) {
+	newPod := newObj.(*apiv1.Pod)
+	podName := newPod.ObjectMeta.Name
+	glog.V(5).Infof("Pod added: %s", podName)
+
+	if newNetworks, ok := newPod.Annotations["networks"]; ok {
+		glog.V(5).Infof("Pod's %s networks annotation '%s' found", podName, newNetworks)
+		nets, err := getNetworks(newNetworks)
+		if err != nil {
+			glog.V(4).Infof("Failed to unmarshall pod's %s new networks annotation, ignore: %s", podName, err)
+			return
+		}
+		np := cniPodNetworkProperty{}
+		for _, n := range nets {
+			np.IfMAC = n.IfMAC
```

Comment:
np is missing isPrimary initialization from n?

Reply:
Confirmed that it was missing. Added